core/src/subgraph/instance_manager.rs

```diff
@@ -841,6 +836,19 @@ async fn process_block<T: RuntimeHostBuilder<C>, C: Blockchain>(
     }
 
     if let Some(proof_of_indexing) = proof_of_indexing {
+        // If a deterministic error has happened, write a new
+        // ProofOfIndexingEvent::DeterministicError to the SharedProofOfIndexing.
+        if has_errors && !is_non_fatal_errors_active {
+            // TODO: get the correct causality region
```

Comment:
What's the status on this TODO?

Reply:
Do you have a suggestion of where that source of truth would be? 🤔

Reply:
Right now this value is constructed (IIRC) in the host export constructor, which I assume is where you copied it from. It would be nice if there was just one place. I don't care where it ends up, but having two places is going to be error prone when we start introducing new regions.

BTW: It turns out there is a nifty way to ensure that only 2 entities need to be queried to construct a PoI once we have causality regions for off-chain data. Not relevant right now, but I just want to mention this so that when the time comes we can talk about it.

Reply:
Oh okay no problem, I'll centralize this then to avoid duplication.

Reply:
I've done it here: https://github.com/graphprotocol/graph-node/pull/2748/commits/34428ac06753ff797d012e18e40969a294c6236b#diff-b3d0b8016fff8abd89acad30bfd8e246bd28b54fc340dd4b98c3a19174da7134R45

I would like an opinion if this is a bad place, since I put it there more to use `CausalityRegion` as a "namespace". Instead of following the Rust standard of methods with the `from_` prefix, that implies a "constructor" of some sort.